typespec-tests/tsp/multi-content-types.tsp

```diff
@@ -19,48 +20,104 @@ model Resource {
   name: string;
 }
 
-@route("/single")
+@route("/single/request")
 interface SingleContentType {
   @doc("response is binary")
   @route("/download/image")
   @get
-  downloadImage(): OkResponse & {
+  downloadImageForSingleContentType(): OkResponse & {
     @header("content-type") contentType: "image/png";
     @body data: bytes;
   };
 
   @doc("request is binary")
   @route("/upload/image")
   @post
-  uploadImage(@body data: bytes, @header contentType: "image/png"): void;
+  uploadImageForSingleContentType(@body data: bytes, @header contentType: "image/png"): void;
 }
 
-@route("multiple")
-interface MultipleContentTypes {
+@route("multiple/sharedroute/request")
+interface MultipleContentTypesOnRequest {
   @doc("one data type maps to multiple content types")
   @route("upload/single-body-type")
   @post
-  uploadBytes(@body data: bytes, @header contentType: "application/octet-stream" | "image/jpeg" | "image/png"): void;
+  uploadBytesWithSingleBodyTypeForMultiContentTypes(
+    @body data: bytes,
+    @header contentType: "application/octet-stream" | "image/jpeg" | "image/png"
+  ): void;
+
+  @doc("multiple data types map to multiple content types using shared route")
+  @route("upload/multi-body-types")
+  @sharedRoute
+  @post
+  uploadBytesWithMultiBodyTypesForMultiContentTypes(
+    @body data: bytes,
+    @header contentType: "application/octet-stream" | "image/jpeg" | "image/png"
+  ): void;
+
+  @doc("multiple data types map to multiple content types using shared route")
+  @route("upload/multi-body-types")
+  @sharedRoute
+  @post
+  uploadJsonWithMultiBodyTypesForMultiContentTypes(@body data: Resource, @header contentType: "application/json"): void;
+
+  @doc("multiple data types map to multiple content types using shared route")
+  @route("upload/multi-body-types")
+  @sharedRoute
+  @post
+  uploadJsonOrBytesWithMultiBodyTypesForMultiContentTypes(
+    @body data: bytes | Resource,
+    @header contentType: "application/json" | "application/octet-stream" | "image/jpeg" | "image/png"
+  ): void;
```

Comment:
I assume typically for sharedRoute case, `uploadBytesWithMultiBodyTypesForMultiContentTypes` and `uploadJsonWithMultiBodyTypesForMultiContentTypes` should already cover the cases of 4 content-types, and hence dev won't write `uploadJsonOrBytesWithMultiBodyTypesForMultiContentTypes`.

Reply:
Yes, just add this edge case to test our behavior because we don't have lint rule to forbid this.
I can remvoe this if you all feel it's unnecessary.

Reply:
Our test should be fine, even if you had it. Be careful on cadl-ranch one.